R/chat.R

```diff
@@ -535,8 +541,127 @@ print.ChatOpenAI <- function(x, ...) {
   invisible(x)
 }
 
+#' @export
+#' @examplesIf elmer:::openai_key_exists()
+#' chat <- new_chat_openai(echo = TRUE)
+#' chat$chat(
+#'   list(
+#'     "What do you see in these images?",
+#'     content_image_url("https://www.r-project.org/Rlogo.png"),
+#'     content_image_file(system.file("httr2.png", package = "elmer"))
+#'   )
+#' )
+content_image_url <- function(url, detail = c("auto", "low", "high")) {
+  # TODO: Allow vector input?
+  check_string(url)
+
+  list(type = "image_url", image_url = list(url = url))
+}
+
+#' @rdname content_image_url
+#' @export
+content_image_file <- function(path, detail = c("auto", "low", "high"), content_type = "auto", resize = NULL) {
+  # TODO: Allow vector input?
+  check_string(path)
+  if (!file.exists(path)) {
+    stop("File does not exist: ", path)
+  }
+
+  if (content_type == "auto") {
+    # OpenAI supports .png, .jpeg, .jpg, .webp, .gif
+    # https://platform.openai.com/docs/guides/vision/what-type-of-files-can-i-upload
+    ext <- tolower(tools::file_ext(path))
+    content_type <- switch(
+      ext,
+      png = "image/png",
+      jpeg = "image/jpeg",
+      jpg = "image/jpeg",
+      webp = "image/webp",
+      gif = "image/gif",
+      stop("Unsupported image file extension: ", ext)
+    )
+  }
+
+  if (!is.null(resize)) {
+    # Load the image
+    img <- magick::image_read(path)
+    img <- magick::image_scale(img, "512x512")
```

Comment:
I think that's the correct size for `detail = "low"`, for detail high, it should be 2048x2048

Reply:
🤦‍♂️  I meant to make it `magick::image_scale(img, resize)`

Reply:
I think you could still pick for the user.